docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,513 @@
+     return list("status", Status.Alive);
+   }
+
+   public void deleteStefs(){
+     delete("name", "Stef");
+  }
+}
+--
+
+Absolutely all the operations that are defined on `PanacheMongoEntityBase` are available on your DAO, so using it
+is exactly the same except you need to inject it:
+
+[source,java]
+--
+@Inject
+PersonRepository personRepository;
+
+@GET
+public long count(){
+  return personRepository.count();
+}
+--
+
+So if Repositories are your thing, you can keep doing them. Even with repositories, you can keep your entities as
+subclasses of `PanacheMongoEntity` in order to get the ID and public fields working, but you can even skip that and
+go back to specifying your ID and using getters and setters if that's your thing. We're not judging.
+
+== Transactions
+
+MongoDB offers ACID transactions since version 4.0. MongoDB with Panache doesn't provides support for them.
+
+== Custom IDs
+
+IDs are often a touchy subject, in MongoDB they are usually auto-generated by the database with an `ObjectId`. 
+In MongoDB with Panache the ID are defined by a field name `id` of the `org.bson.types.ObjectId` type,
+but if you want ot customize them, once again we have you covered.
+
+You can specify your own ID strategy by extending `PanacheMongoEntityBase` instead of `PanacheMongoEntity`. Then
+you just declare whatever ID you want as a public field by annotating it by @BsonId:
+
+[source,java]
+--
+@MongoEntity
+public class Person extends PanacheMongoEntityBase {
+
+  @BsonId
+  public Integer myId;
+
+ ...
+}
+--
+
+If you're using repositories, then you will want to extend `PanacheMongoRepositoryBase` instead of `PanacheMongoRepository`
+and specify your ID type as an extra type parameter:
+
+[source,java]
+--
+@ApplicationScoped
+public class PersonRepository implements PanacheMongoRepositoryBase<Person,Integer> {
+
+ ...
+}
+--
+
+[NOTE]
+====
+When using `ObjectId`, MongoDB will automatically provide a value for you, but if you use a custom field type,
+you need to provides the value by yourself.
+====
+
+`ObjectId` can be difficult to use if you want to expose its value in your rest service.
+So we create JsonB providers to serialize/deserialize them as String.
+These providers are not registered by default so you need to register them with a JAXRS `ContextResolver`:
+
+[source,java]
+--
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+import javax.json.bind.JsonbConfig;
+import javax.ws.rs.ext.ContextResolver;
+import javax.ws.rs.ext.Provider;
+
+import io.quarkus.mongodb.panache.jsonb.ObjectIdDeserializer;
+import io.quarkus.mongodb.panache.jsonb.ObjectIdSerializer;
+
+@Provider
+public class QuarkusJsonbContextResolver implements ContextResolver<Jsonb> {
+
+    public Jsonb getContext(Class clazz) {
+        JsonbConfig config = new JsonbConfig();
+        config.withSerializers(new ObjectIdSerializer()).withDeserializers(new ObjectIdDeserializer());
+        return JsonbBuilder.create(config);
+    }
+
+}
```

Comment:
It seems a bit unfortunate that you have to manually add this for the default id behavior.

Reply:
Agreed, it's on the TODO list on the description of the PR :)

But the discussion about how to know if an extension is included needs to more forward because today there is no way to know if JsonB is included (not all extension provides a Capabilities - in fact only Arc and Narayana, so therer is currently no way to know that an extension is included).

Reply:
Well, just add a capability if you need one. You can do that in separate PRs, we will merge them.

Reply:
OK, will do

Reply:
In fact, it's more complex that I first thought ...

Moreover the PR #3553, if merged, will change the way serializers/deserializers will be managed.
For the moment, I will update the TODO list with to do it later

Reply:
@gsmet both JSON-B and Jackson serializer/deserializer are automatically registered if resteasy-jsonb or resteasy-jackson capability is present.

When changing one of these two library we need to carefully review this as it may creates incompabilities (for example an impact analyses needs to be done for PR #3553, and if we add some customization to Jackson ObjectMapper we need to port them to panache-mongo or change the way the ObjectMapper is created).

Fo this to work, I had to create a `resteasy-common-spi` module.